go/tools/gazelle/packages/package.go

```diff
@@ -15,7 +15,46 @@ limitations under the License.
 
 package packages
 
-import "strings"
+import (
+	"fmt"
+	"strings"
+)
+
+// PlatformConstraints is a map from config_setting labels (for example,
+// "@io_bazel_rules_go//go/platform:linux_amd64") to a sets of build tags
+// that are true on each platform (for example, "linux,amd64").
+type PlatformConstraints map[string]map[string]bool
+
+// ParsePlatformConstraints parses a string that contains all platforms and
+// tags on each platform. The string must contain an even number of fields,
+// separated by whitespace. Even fields (counting from 0) are platform labels.
```

Comment:
Part of my worry is this...
	android	arm
	darwin	386
	darwin	amd64
	darwin	arm
	darwin	arm64
	dragonfly	amd64
	freebsd	386
	freebsd	amd64
	freebsd	arm
	linux	386
	linux	amd64
	linux	arm
	linux	arm64
	linux	ppc64
	linux	ppc64le
	linux	mips
	linux	mipsle
	linux	mips64
	linux	mips64le
	netbsd	386
	netbsd	amd64
	netbsd	arm
	openbsd	386
	openbsd	amd64
	openbsd	arm
	plan9	386
	plan9	amd64
	solaris	amd64
	windows	386
	windows	amd64
Is the full list of goos/goarch pairs that make up a platform, anything that asks a user to repeat that list on a command line is probably a broken user interface, especially as it's hard coded in the go tools.
I think we should presume that full set exists, and just list the additional tags that might apply, and assume they apply to all those combinations.
You can even pick the GOOS and GOARCH lists out of go/build/syslist.go and just assume all combinations are valid.